ansible/roles-infra/infra-osp-project-create/tasks/main.yaml

```diff
@@ -1,79 +1,118 @@
 ---
-- name: Create project and quota
-  when: osp_project_create
+- name: Generate user password
+  set_fact:
+    heat_user_password_generated: "{{lookup('password', '/dev/null length=12 chars=ascii_letters,digits')}}"
+
+- name: Set heat_user_password
+  set_fact:
+    heat_user_password: "{{ heat_user_password_provided | default(heat_user_password_generated) }}"
+
+- name: Set fact for new user password
+  set_fact:
+    osp_auth_password_member: "{{ heat_user_password_provided | default(heat_user_password_generated) }}" 
```

Comment:
equivalent to the simpler:

```yaml
 osp_auth_password_member: "{{ heat_user_password }}"
```

Is that what we want ?  If i want to provide `osp_auth_password_member`, i need to actually set:

* `osp_auth_password_member`
* `heat_user_password`

That sounds strange to me.

Also, now, the user is not created in the Heat template, so there is no reason to keep `heat_user_password`, right ?

We should just replace every occurence of `heat_user_password` with `osp_auth_password_member`. Wdyt ?

i checked, `heat_user_password` is not used in agnosticv.\

In agnosticd:

```
[fridim@lolo agnosticd]$ grep heat_user_password * -r
ansible/roles-infra/infra-osp-template-generate/templates/nested_version/cloud_template_master.j2:      password: {{ heat_user_password }}
ansible/roles-infra/infra-osp-template-generate/templates/cloud_template_master.j2:      password: {{ heat_user_password }}
ansible/roles-infra/infra-osp-template-generate/tasks/generate_osp_template.yml:    heat_user_password_generated: "{{lookup('password', '/dev/null length=12 chars=ascii_letters,digits')}}"
ansible/roles-infra/infra-osp-template-generate/tasks/generate_osp_template.yml:    heat_user_password: "{{ heat_user_password_provided | default(heat_user_password_generated) }}"
ansible/roles/host-ocp4-provisioner/templates/clouds.yaml.j2:      password: "{{ hostvars['localhost']['heat_user_password'] }}"
ansible/configs/rhel8lab/post_software.yml:          bookbag_os_password: "{{ hostvars.localhost.heat_user_password }}"
ansible/configs/ansible-tower-implementation/files/cloud_providers/osp_cloud_template_master.j2:      password: {{ heat_user_password }}
ansible/configs/sap-rhvh/files/cloud_providers/osp_cloud_template_master.j2:      password: {{ heat_user_password }}
ansible/configs/ocp4-disconnected-osp-lab/software.yml:        #     OS_PASSWORD: "{{ hostvars['localhost']['heat_user_password'] }}"
ansible/configs/ocp4-disconnected-osp-lab/files/clouds.yaml.j2:      password: "{{ hostvars['localhost']['heat_user_password'] }}"
ansible/configs/osp-sandbox/files/clouds.yaml.j2:      password: "{{ hostvars['localhost']['heat_user_password'] }}"
ansible/configs/osp-sandbox/files/secret.yaml.j2:osp_auth_password: "{{ hostvars['localhost']['heat_user_password'] }}"
ansible/configs/osp-sandbox/post_software.yml:        - "user.info:       password: {{ hostvars['localhost']['heat_user_password'] }}"
ansible/configs/sap-hana/files/cloud_providers/osp_cloud_template_master.j2:      password: {{ heat_user_password }}
ansible/configs/multi-cloud-capsule/files/cloud_providers/osp_cloud_template_master.j2:      password: {{ heat_user_password }}```

Reply:
I kept `heat_user_password` in here really for compatibility and to minimize change since this is used elsewhere. Ultimately we should no longer need it and `osp_auth_password_member` should be sufficient, but the changes in this PR were already getting bigger than I wanted.

I'd prefer to keep both as is and I will add a comment to `heat_user_password` that it is being deprecated and will be removed later?

Reply:
OK, let's track this in an issue.